### ISSUE TITLE
Fil the Junit View while running a maven build interactivly

### ISSUE THREAD

**Issue by laeubi** [eclipse-m2e/m2e-core#1023]
Currently when running a maven build "nothing" happens (you see success/failed of course). But m2e actually can know where such files are produce and setup a FileWatcher and then redirect the written results to the JUnit view.

**Comment by HannesWell** [comment]
Since M2E-1.20 for maven projects with a pom.xml  you already get a link in the console to test results, that opens the JUnit view:
https://github.com/eclipse-m2e/m2e-core/blob/master/RELEASE_NOTES.md#improved-links-to-junit-test-reports-and-project-pomxml-in-the-console-of-a-maven-build

For polyglot projects this will work as well once the following PR is merged:
- https://github.com/eclipse-m2e/m2e-core/pull/572 is merged.

But of course that does not show the results live. Nevertheless the infrastructure build with #572 will probably help to realize that because it builds a connection between the Maven build VM and the Eclipse/M2E VM and therefore makes M2E aware where the test results are written.
The approach is inspired by the M2E MavenDev tools from Igor Fedorenko:
https://github.com/ifedorenko/com.ifedorenko.m2e.mavendev

Among other useful things they come with a live view that interactively shows the executed and planned goals of a launched build. Something similar could be done for the JUnit view.

Furthermore the files opened from the links on the console only show the result for one module/project. For an entire Maven-build the Junit-view could show an aggregate, where each (non-pom) module is one top level element so that one get an aggregated result.

**Comment by laeubi** [comment]
You dont really need a connection to the vm, just add a Directorywatcher for the basedir of the executed maven and you will be notified about new files, then it is jsut a matter of check if it is a surefire result, aggregate new files in a result and show it in the view. I even used the new JUnitview once to show "live" results, but if the view is jsut updated I think thats perfectly suitable.

**Comment by akurtakov** [comment]
I would recommend not using JUnitView but rather the generic TestUnit view from https://github.com/eclipse-platform/eclipse.platform.debug/tree/master/org.eclipse.unittest.ui . This is what was created to stop every plugin creating their own unit view (Corrosion, WWD, JDT... already support it) but it still lacks general acceptance .

**Comment by laeubi** [comment]
@akurtakov yes that's what I mean by "new JUnitView" I already used that in the past for the cucumber-eclipse project and it works quite nice.

**Comment by laeubi** [comment]
Found it: https://github.com/cucumber/cucumber-eclipse/blob/90195bf4bff2721c6d5a11f37865f63b445cc267/io.cucumber.eclipse.editor/src/io/cucumber/eclipse/editor/testresults/CucumberTestRunnerClient.java

this uses a remote process connection to watch test-execution and publish it inside eclipse.

**Comment by stbischof** [comment]
Would be great is you could Aggregate several xml files together into one view.

With Filewatchers you could dynamic Look.
Simpler wax would be just at end if run compare the timestamp.
